tests/TestCase/ORM/QueryTest.php

```diff
@@ -3548,4 +3562,142 @@ public function testNotMatchingNested()
+        $this->getTableLocator()->get('Articles');
+        $table->hasMany('Comments');
+        $table->belongsTo('Authors');
+
+        $result = $table
+            ->find()
+            ->contain([
+                'Comments' => function ($query) use ($table) {
+                    return $query->selectAllExcept($table->Comments, ['published']);
+                }
+            ])
+            ->selectAllExcept($table, ['body'])
+            ->first();
+        $this->assertNull($result->comments[0]->published);
+        $this->assertNull($result->body);
+        $this->assertNotEmpty($result->id);
+        $this->assertNotEmpty($result->comments[0]->id);
+    }
+
+    /**
+     * Test what happens if you call selectAllExcept() more
+     * than once.
+     *
+     * @return void
+     */
+    public function testSelectAllExceptWithMulitpleCalls()
+    {
+        $table = $this->getTableLocator()->get('Articles');
+        $this->getTableLocator()->get('Articles');
+
+        $result = $table
+            ->find()
+            ->selectAllExcept($table, ['body'])
+            ->selectAllExcept($table, ['published']);
+        $selectedFields = $result->clause('select');
+        $expected = [
+            'Articles__id' => 'Articles.id',
+            'Articles__author_id' => 'Articles.author_id',
+            'Articles__title' => 'Articles.title',
+            'Articles__published' => 'Articles.published',
+            'Articles__body' => 'Articles.body'
+        ];
+        $this->assertEquals($expected, $selectedFields);
+
+        $result = $table
+            ->find()
+            ->selectAllExcept($table, ['body'])
+            ->selectAllExcept($table, ['published', 'body']);
+        $selectedFields = $result->clause('select');
+        $expected = [
+            'Articles__id' => 'Articles.id',
+            'Articles__author_id' => 'Articles.author_id',
+            'Articles__title' => 'Articles.title',
+            'Articles__published' => 'Articles.published'
+        ];
+        $this->assertEquals($expected, $selectedFields);
+
+        $result = $table
+            ->find()
+            ->selectAllExcept($table, ['body'])
+            ->selectAllExcept($table, ['published', 'body'], true);
+        $selectedFields = $result->clause('select');
+        $expected = [
+            'Articles__id' => 'Articles.id',
+            'Articles__author_id' => 'Articles.author_id',
+            'Articles__title' => 'Articles.title',
+        ];
+        $this->assertEquals($expected, $selectedFields);
+    }
+
+    /**
+     * Test that given the wrong first parameter, Invalid argument exception is thrown
+     *
+     * @return void
+     */
+    public function testSelectAllExceptThrowsInvalidArgument()
+    {
+        $table = $this->getTableLocator()->get('Articles');
+        try {
+            $table
+                ->find()
+                ->selectAllExcept([], ['body']);
+        } catch (\InvalidArgumentException $e) {
```

Comment:
We generally use `expectException(InvalidArgumentException::class)`

Reply:
Thanks for the pro-tip.